utils/bigclown/bigclown-gateway/files/init

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh /etc/rc.common
+
+START=98
+STOP=1
+
+USE_PROCD=1
+
+PROG=/usr/bin/bcg
+CONF=/tmp/etc/bigclown-gateway.conf
+
+append() {
+	local cfg="$1"
+	local uci_name="$2"
+	local out_name="$3"
+	local default="$4"
+	config_get val $cfg $uci_name $default
+	if [ -n "$val" ]; then
+	  echo "$out_name $val" >> $CONF
+	fi
+}
+
+start_service() {
+	config_load bigclown-gateway
+
+	rm -rf $CONF
+	echo "Generating bigclown-gateway config file in $CONF"
+
+	append gateway name 'name:' usb-dongle
+	# TODO add udev rules and use different default here
+	append gateway device 'device:' /dev/ttyUSB0
+	append gateway automatic_rename_kit_nodes 'automatic_rename_kit_nodes:' 1
+	append gateway base_topic_prefix 'base_topic_prefix:'
+
+	echo "mqtt:" >> $CONF
+	append mqtt host '  host:' localhost
+	append mqtt port '  port:' 1883
+	append mqtt cafile '  cafile:'
+	append mqtt certfile '  certfile:'
+	append mqtt keyfile '  keyfile:'
+
+	procd_open_instance
```

Comment:
do you perhaps want to have a UCI option that controls enablign/disabling the service? that lets you have sample config values, but it won't try and automatically start doign things that with default configs.  Similar to how wireless is disabled by default.

Reply:
I added one.